samtranslator/swagger/swagger.py

```diff
@@ -534,6 +534,14 @@ def set_path_default_authorizer(
                 for method_definition in self.get_method_contents(self.get_path(path)[normalized_method_name]):
 
                     # If no integration given, then we don't need to process this definition (could be AWS::NoValue)
+                    if not isinstance(method_definition, dict):
```

Comment:
Can we start refactoring these branches? The patterns are essentially identical, but takes 8 LoC for a single check.

Reply:
irrelevant: we should have mypy to end this manual type check nightmare!

Reply:
@hoffa I dont understand. What (or how) do you mean refactor?
@aahung These errors are coming from user template. How will mypy know what is valid and what is not?

Reply:
if we define the possible types of `method_definition`, and mypy will scream if you don't have this check.

Reply:
This is the challenge right now and itself could be a lot of work (of course worth doing) which is to assess and note every parameter's data type. If we can do that then it would solve all current and future SAM crashes as well.

Reply:
@prenx4x `grep InvalidDocumentException . -RC2` gives me a bunch of very similar checks: if value doesn't match type, throw the exception. It's noisy and repetitive, would be good to start refactoring these so we can simplify common patterns and reduce cognitive load, otherwise it won't be maintainable.

Reply:
And more generally, this sort of "lazy" checking doesn't seem to be ideal; maybe we can validate up-front what needs to be validated, and use [EAFP](https://docs.python.org/3.9/glossary.html#term-eafp) elsewhere. This would reduce complexity and improve maintainability.

Reply:
I see. One thing to note here is that the errors could be similar, but the message (or cause) is very different. This message is seen by the user and it should be clear enough to point out the issue in their template. We we find same errors with same cause at multiple places we can refactor it, but if causes are different then they should be separate exceptions.

Reply:
Right, and that's fine; messages can be custom. But we should still be continuously improving our codebase.

This doesn't mean refactoring everything at once, but if we recognize common patterns that can benefit from refactoring in _new_ code, we should go ahead and do it, and ensure that in the long term our quality keeps increasing.